modules/core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/QueryMakerServiceImpl.java

```diff
@@ -20,88 +20,43 @@
 
 import java.util.ArrayList;
 import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
+import java.util.stream.Collectors;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
 import org.nuxeo.ecm.core.query.sql.NXQL;
-import org.nuxeo.runtime.model.ComponentInstance;
+import org.nuxeo.runtime.model.ComponentContext;
 import org.nuxeo.runtime.model.DefaultComponent;
 
 /**
  * Service for the registration of QueryMaker classes.
  */
 public class QueryMakerServiceImpl extends DefaultComponent implements QueryMakerService {
 
-    private static final Log log = LogFactory.getLog(QueryMakerServiceImpl.class);
-
     public static final String XP = "queryMaker";
 
-    protected final List<QueryMakerDescriptor> descriptors = new ArrayList<>(2);
-
     protected List<Class<? extends QueryMaker>> queryMakers;
 
     @Override
-    public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
-        if (XP.equals(extensionPoint)) {
-            registerQueryMaker((QueryMakerDescriptor) contribution);
-        }
-    }
-
-    @Override
-    public void unregisterContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
-        if (XP.equals(extensionPoint)) {
-            unregisterQueryMaker((QueryMakerDescriptor) contribution);
+    public void start(ComponentContext context) {
+        queryMakers = new ArrayList<>(2);
+        queryMakers.addAll(this.<QueryMakerDescriptor> getRegistryContributions(XP)
```

Comment:
Could probably assign directly. `Collectors.toList` in practice returns an `ArrayList` and this won't change (as too many people already rely on that, per OpenJDK dev discussions).